app/src/main/java/org/fossasia/openevent/general/event/topic/SimilarEventsViewModel.kt

```diff
@@ -16,20 +17,24 @@ class SimilarEventsViewModel(private val eventService: EventService) : ViewModel
     val progress = MutableLiveData<Boolean>()
     val similarEvents = MutableLiveData<List<Event>>()
     val error = MutableLiveData<String>()
+    var eventList = ArrayList<Event>()
+    var eventId: Long = -1
 
     fun loadSimilarEvents(id: Long) {
         if(id == -1L){
             return
         }
         compositeDisposable.add(eventService.getSimilarEvents(id)
                 .subscribeOn(Schedulers.io())
+                .flatMapIterable{ it }
+                .filter { it.id != eventId }
```

Comment:
Add toList

Reply:
You have to maintain a new list because of it, and new trigger will duplicate items. Logic is wrong